lib/versioning/loose/generic.ts

```diff
@@ -39,41 +49,37 @@ export abstract class GenericVersioningApi<
     const length = Math.max(left.release.length, right.release.length);
     for (let i = 0; i < length; i += 1) {
       // 2.1 and 2.1.0 are equivalent
-      const part1 = left.release[i] ?? 0;
-      const part2 = right.release[i] ?? 0;
-      if (part1 !== part2) {
-        return part1 - part2;
+      const leftPart = left.release[i] ?? 0;
+      const rightPart = right.release[i] ?? 0;
+      if (leftPart < rightPart) {
+        return -1;
+      } else if (leftPart > rightPart) {
+        return 1;
       }
     }
 
-    if (
-      is.nonEmptyString(left.prerelease) &&
-      is.nonEmptyString(right.prerelease)
-    ) {
-      const pre = left.prerelease.localeCompare(right.prerelease);
-
-      if (pre !== 0) {
-        return pre;
-      }
-    } else if (is.nonEmptyString(left.prerelease)) {
+    const leftLength = left.release.length;
+    const rightLength = right.release.length;
+    if (leftLength < rightLength) {
```

Comment:
Every implementation based on `loose` uses it's own `_compare`, so sticking to the logic from `index.ts` (which is being leveraged in our buildpack repos)

Reply:
Where we use loose in buildpack? 🤔

Reply:
Seems like I was wrong, though found this place:
https://github.com/renovatebot/docker-swift/blob/1adf2c773c1bf7c4836a0e750a80cfa78d8309c8/Dockerfile#L1

Reply:
Just aiming to make this code work the same way as before, for those who imported from `index.ts`